### PR TITLE
Refine cloud provider docs

### PR DESCRIPTION
**Pull request by feiskyer**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
/kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: Refine cloud provider docs

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Release note**:
```
NONE
```

### CODE REVIEW

File: docs/cloud-provider-config.md [hunk 1]
```diff
@@ -287,7 +285,6 @@ For authentication methods, only Service Principal supports this feature, and `a
 The following are the default rate limiting values configured in [AKS](https://azure.microsoft.com/en-us/services/kubernetes-service/) and [AKS-Engine](https://github.com/Azure/aks-engine) clusters prior to Kubernetes version v1.18.0.
 
 ```json
-    "cloudProviderBackoffMode": "v2",
```
Comment (ritazh) [inline docs/cloud-provider-config.md:290]: Can you add more details in the table to when `v2` should be specified?

Comment (feiskyer) [reply docs/cloud-provider-config.md:290]: "v2" is not supported anymore since v1.18 as cloud-provider's own backoff retries have been added.

Comment (feiskyer) [reply docs/cloud-provider-config.md:290]: Also, AKS doesn't set this option

Comment (ritazh) [reply docs/cloud-provider-config.md:290]: Can you add this info to the table as such?

Comment (feiskyer) [reply docs/cloud-provider-config.md:290]: Good catch, added now.

File: docs/cloud-provider-config.md [hunk 2]
```diff
@@ -74,7 +74,7 @@ If more than one value is set, the order is `Managed Identity` > `Service Princi
 |cloudProviderBackoffExponent|Backoff exponent|Float value, valid if `cloudProviderBackoff` is true|
 |cloudProviderBackoffDuration|Backoff duration|Integer value, valid if `cloudProviderBackoff` is true|
 |cloudProviderBackoffJitter|Backoff jitter|Float value, valid if `cloudProviderBackoff` is true|
-|cloudProviderBackoffMode|Backoff mode, supported values are "v2" and "default"|Default to "default"|
+|cloudProviderBackoffMode|Backoff mode, supported values are "v2" and "default". Note that "v2" has been depracated since v1.18.0. |Default to "default"|
```
Comment (ritazh) [inline docs/cloud-provider-config.md:77]: nit: deprecated

**Pull request merged**